### PR TITLE
Fix: utxo-lib transaction weight

### PR DESCRIPTION
**Pull request by szymonlesisz**
Refactored transaction estimated (virtual) bytes

- constants (input/output bytes) are moved into 1 place (coinselect module)
- virtual bytes are calculated from tx weight
- tx weight calculation is inspired by [trezor-firmware/TxWeightCalculator](https://github.com/trezor/trezor-firmware/blob/1fceca73da523c5bf2bb0f398c91e00c728bdbe0/core/src/apps/bitcoin/sign_tx/tx_weight.py)
- composeTx tests results are verified with [TxWeightCalculator](https://github.com/trezor/trezor-suite/blob/094c4399bd082538d43b73b5b792647d4d05eb62/packages/utxo-lib/tests/compose.utils.ts) ported to javascript.
- adjusted fixtures (lower fees)

### CODE REVIEW

**Comment by szymonlesisz** [comment]
tagging @andrewkozlik as well,
you probably want to see [this](https://github.com/trezor/trezor-suite/blob/be202938fc0a60c150580f64ab1fa19c679daee4/packages/utxo-lib/src/coinselect/utils.ts#L13-L93) tx weight calc. and [this](https://github.com/trezor/trezor-suite/blob/094c4399bd082538d43b73b5b792647d4d05eb62/packages/utxo-lib/tests/compose.utils.ts) TxWeightCalculator in javascript

File: packages/utxo-lib/tests/compose.utils.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+// size of a p2wsh script (1 version, 1 push, 32 hash)
+const _TXSIZE_WITNESSSCRIPT = 34;
+
+// const OUTPUT_SCRIPT_LENGTH = {
+//     p2pkh: 25,
+//     p2sh: 23,
+//     p2tr: 34,
+//     p2wpkh: 22,
+//     p2wsh: 34,
+// };
```
Comment (szymonlesisz) [inline packages/utxo-lib/tests/compose.utils.ts]: odmazat

File: packages/utxo-lib/tests/compose.utils.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+        return total;
+    }
+
+    getVirtualBytes() {
+        return Math.ceil(this.getTotal() / 4);
+    }
+}
+
+export function verifyTxBytes(tx: any, txType: keyof typeof SCRIPT_TYPES = 'p2pkh', network?: any) {
```
Comment (szymonlesisz) [inline packages/utxo-lib/tests/compose.utils.ts]: type

File: packages/utxo-lib/tests/compose.utils.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    witness?: Buffer[];
+    ownership_proof?: any;
+};
+
+export class TxWeightCalculator {
+    inputs_count = 0;
+    outputs_count = 0;
+    counter = 4 * (_TXSIZE_HEADER + _TXSIZE_FOOTER);
+    segwit_inputs_count = 0;
+    inputs: any[] = [];
```
Comment (szymonlesisz) [inline packages/utxo-lib/tests/compose.utils.ts]: typ

**Review by mroz22 — APPROVED** [review]

File: packages/utxo-lib/tests/compose.utils.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+// transaction segwit overhead 2 (marker, flag)
+const _TXSIZE_SEGWIT_OVERHEAD = 2;
+
+// transaction input size (without script): 32 prevhash, 4 idx, 4 sequence
+const _TXSIZE_INPUT = 40;
+// transaction output size (without script): 8 amount
+const _TXSIZE_OUTPUT = 8;
+// size of a pubkey
+const _TXSIZE_PUBKEY = 33;
+// average size of a DER signature (3 type bytes, 3 len bytes, 32.5 R, 32.5 S, 1 sighash)
```
Comment (andrewkozlik) [inline packages/utxo-lib/tests/compose.utils.ts]: Please fix this comment as per https://github.com/trezor/trezor-firmware/pull/1891/commits/ea8686aff4a75e93998a28002fd7780de6a5c168 (my bad).

**Review by andrewkozlik — APPROVED** [review]
The weight calculation algorithm LGTM.

File: packages/utxo-lib/tests/compose.utils.ts (hunk 4 above)
Comment (szymonlesisz) [reply packages/utxo-lib/tests/compose.utils.ts]: https://github.com/trezor/trezor-suite/pull/4482/commits/dd5b8e5d5bf6c71445d3c802ed436acd69d3dc68

**Pull request merged**